app/controllers/root_controller.rb

```diff
@@ -273,10 +259,47 @@ def treat_as_standard_html_request?
     !request.format.json? and !request.format.print? and !request.format.video?
   end
 
-  def local_transaction_details(artefact, authority_slug, snac)
-    raise RecordNotFound if snac.blank? && authority_slug.present?
+  def local_transaction_details(artefact, authority_slug)
+    if authority_slug
+      lgsl = artefact['details']['lgsl_code']
+      lgil = artefact['details']['lgil_override']
 
-    artefact['details'].slice('local_authority', 'local_service', 'local_interaction')
+      local_links_manager_response = local_links_manager_request(authority_slug, lgsl, lgil)
+      artefact['details'].slice('local_service').merge(local_links_manager_response)
+    else
+      artefact['details'].slice('local_service')
+    end
+  end
+
+  def local_links_manager_request(authority_slug, lgsl, lgil)
+    # RESTful endpoint
+    # url = "/local_authorities/#{authority_slug}/services/#{lgsl}/interactions/#{lgil}.json"
+
+    # Non-RESTful endpoint
+    url = "/api/local_interaction_details.json?authority_slug=#{authority_slug}&lgsl=#{lgsl}&lgil=#{lgil}"
+
+    # There will be a wrapper in gds-api-adapters for LocalLinksManager
+    # LocalLinksManager.get(url)
+
+    # this is the hash that was previously added to the artefact['details'] after
+    # the artfact was updated with local_transaction details by calling
+    # `fetch_artefact`
+    # Instead we return this hash from LocalLinksManager's API endpoint
+
+    # For demonstration purposes we return this stubbed response
+    {
+      "local_authority"=>{
+        "name"=>"London Borough of Camden",
+        # "snac"=>"00AG",
+        # "tier"=>"unitary",
+        "homepage_url"=>"http://www.camden.gov.uk/"
+      },
+      "local_interaction"=>{
+        # "lgsl_code"=>432,
+        # "lgil_code"=>0,
+        "url"=>"https://www.westminster.gov.uk/stray-and-lost-dogs"
+      }
```

Comment:
We've taken out one MapIt request [here](https://github.com/alphagov/frontend/pull/977/files/15ed19e09e28695d5c592d1331f45a5165385137#diff-132a19fd5e7a1edc0d9127204dfa48eaL138), which converted 'slug' to 'snac', which we don't need anymore as we don't use snacs for LocalInteraction lookups (and decided that providing support for the `/api/` route wasn't necessary). So the only MapIt request remaining would be the one from the initial postcode lookup I think. But yeah, let's have another think about the approach for sure!